crates/astria-sequencer/src/app.rs

```diff
@@ -77,6 +77,13 @@ type InterBlockState = Arc<StateDelta<Snapshot>>;
 pub(crate) struct App {
     state: InterBlockState,
 
+    // set to true when `prepare_proposal` is called, indicating we are the proposer for this slot.
+    // set to false when `commit` is called, finalizing the block.
+    // when this is se to true, `process_proposal` is not executed, as we have already executed
+    // the transactions for the block during `prepare_proposal`, and re-executing them would
+    // cause failure.
+    is_proposer: bool,
```

Comment:
yeah, i'll change this to match on the block hash instead

Reply:
yeah i agree, this wouldn't work in that case. i updated this to always set `is_proposer = false` when `process_proposal` is called, so `process_proposal` will skip only in the case where `prepare_proposal` is called directly before it. so if we propose a block, but it's not accepted, and then `process_proposal` is called for the next proposed block, it will execute as normal